### ISSUE TITLE
No-redeclare error: Eslint flag local variables with same name but in different scopes as re-declared.

### ISSUE THREAD

**Issue by leewaygroups** [eslint/eslint#7001]
**What version of ESLint are you using?**
0.24.1

**What parser (default, Babel-ESLint, etc.) are you using?**
gulp-eslint

**Please show your full configuration:**
```js
{
  "ecmaFeatures": {
    "jsx": true
  },
  "env": {
    "browser": true,
    "node": true,
    "jquery": true
  },
  "rules": {
    "quotes": 0,
    "no-trailing-spaces": 0,
    "eol-last": 0,
    "no-unused-vars": 0,
    "no-underscore-dangle": 0,
    "no-alert": 0,
    "no-console": 0,
    "no-lone-blocks": 0
  },
  "globals": {
    "jQuery": true,
    "$": true
  }
}
```

**What did you do? Please include the actual source code causing the issue.**
```js
 if (this.props.style) {
        for (var key in this.props.style) {
          modalStyle[key] = this.props.style[key];
        }
  }

if (this.props.backdropStyle) {
        for (var key in this.props.backdropStyle) {
          backdropStyle[key] = this.props.backdropStyle[key];
        }
}
```
**What did you expect to happen?**
No error should be flagged.

**What actually happened? Please include the actual, raw output from ESLint.**
Raw output from eslint:  52:17  error  key is already defined  no-redeclare

**Comment:**
Apparently, eslint fails to see "key" as two different local variables scoped within the blocks they were individually declared.

**Comment by ilyavolodin** [comment]
JavaScript is a function scoped language (with exception of `let` and `const`). So this rule is behaving correctly and notifies you that hoisted variable is being re-declared.

**Comment by mysticatea** [comment]
Closing as this is correct behavior according to the language spec.
Please use `let`/`const` declarations for block scoping.
